pubsub/acks_test.go

```diff
@@ -297,3 +298,34 @@ func TestSubShutdownCanBeCanceledEvenWithHangingSendAcks(t *testing.T) {
 		t.Fatalf("waited too long (%v) for Shutdown to run", tooLong)
 	}
 }
+
+func TestReceiveReturnsErrorFromSendAcks(t *testing.T) {
+	ctx := context.Background()
+	serr := errors.New("SendAcks failed")
+	ackChan := make(chan struct{})
+	ds := &ackingDriverSub{
+		q: []*driver.Message{
+			&driver.Message{AckID: 0},
+			&driver.Message{AckID: 1},
+		},
+		sendAcks: func(context.Context, []driver.AckID) error {
+			close(ackChan)
+			return serr
+		},
+	}
+	sub := pubsub.NewSubscription(ds)
+	defer sub.Shutdown(ctx)
+	m, err := sub.Receive(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	m.Ack()
+	// Wait for the ack to be sent.
+	<-ackChan
+	// Wait a little longer for the logic after SendAcks returns to happen.
```

Comment:
This seems potentially flaky... could it maybe be updated to call `Receive` until it gets an error?